From 39998d7fa18b8e3fb0edda2d6fb0d38bc83951fb Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Fri, 3 Oct 2014 21:26:12 -0400
Subject: [powerpc] pseries/hvcserver: Fix endian issue in hvcs_get_partner_info

Message-id: <20141003212611.20407.51264.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 96897
O-Subject: [PATCH RHEL7.1 BZ1142244 3/4] powerpc/pseries/hvcserver: Fix endian issue in hvcs_get_partner_info
Bugzilla: 1142244
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1142244

Description:
------------
A buffer returned by H_VTERM_PARTNER_INFO contains device information
in big endian format, causing problems for little endian architectures.
This patch ensures that they are in cpu endian.

Signed-off-by: Thomas Falcon <tlfalcon@linux.vnet.ibm.com>
Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=587870e8650a0571e895cc879cd895c78c6391bf

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/hvcserver.c b/arch/powerpc/platforms/pseries/hvcserver.c
index 4557e91..eedb645 100644
--- a/arch/powerpc/platforms/pseries/hvcserver.c
+++ b/arch/powerpc/platforms/pseries/hvcserver.c
@@ -163,8 +163,8 @@ int hvcs_get_partner_info(uint32_t unit_address, struct list_head *head,
    return retval;
   }
 
-  last_p_partition_ID = pi_buff[0];
-  last_p_unit_address = pi_buff[1];
+  last_p_partition_ID = be64_to_cpu(pi_buff[0]);
+  last_p_unit_address = be64_to_cpu(pi_buff[1]);
 
   /* This indicates that there are no further partners */
   if (last_p_partition_ID == ~0UL
-- 
1.7.1